infrastructure/src/main/java/org/corfudb/infrastructure/management/HealingDetector.java

```diff
@@ -0,0 +1,233 @@
+package org.corfudb.infrastructure.management;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.clients.BaseClient;
+import org.corfudb.runtime.clients.IClientRouter;
+import org.corfudb.runtime.exceptions.WrongEpochException;
+import org.corfudb.runtime.view.Layout;
+
+/**
+ * HealingDetector polls all the "unresponsive members" in the layout.
+ * Unresponsive members: All endpoints in the unresponsiveServers list in the layout.
+ * For every poll method call invoked, it starts a polling round and generates a poll report.
+ * Each polling round comprises of "detectionThreshold" number of iterations.
+ * - We asynchronously poll every known unresponsive member in the layout.
+ * - Poll result aggregation.
+ * - If we complete an iteration without detecting any healed nodes, we end the round.
+ * - Else we continue polling and generate the report with the healed node.
+ * The management Server ensures only one instance of this class and hence this is NOT thread safe.
```

Comment:
I like the not thread safe disclaimer but I still am not a big fan of object scoped variables unless really making sense.